### PR TITLE
Update API comment for nodeName to match system behavior

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

In light of https://github.com/kubernetes/enhancements/issues/3521#issuecomment-2016805887, updating API documentation to reflect the system behavior.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -3667,9 +3667,9 @@ type PodSpec struct {
 	// +optional
 	AutomountServiceAccountToken *bool `json:"automountServiceAccountToken,omitempty" protobuf:"varint,21,opt,name=automountServiceAccountToken"`
 
-	// NodeName is a request to schedule this pod onto a specific node. If it is non-empty,
-	// the scheduler simply schedules this pod onto that node, assuming that it fits resource
-	// requirements.
+	// NodeName indicates in which node this pod is scheduled.
+	// If empty, this pod is a candidate for scheduling by the scheduler defined in schedulerName.
+	// If it is non-empty, the kubelet for this node becomes responsible for the lifecycle of this pod.
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/types.go:3672]: we could also mention using nodeSelector and affinity.nodeAffinity to target specific nodes while still allowing scheduling to work.

Comment (alculquicondor) [reply staging/src/k8s.io/api/core/v1/types.go:3672]: Could I just link to the website documentation instead?

File: pkg/apis/core/types.go [hunk 2]
```diff
@@ -3277,9 +3277,9 @@ type PodSpec struct {
 	// +optional
 	AutomountServiceAccountToken *bool
 
-	// NodeName is a request to schedule this pod onto a specific node.  If it is non-empty,
-	// the scheduler simply schedules this pod onto that node, assuming that it fits resource
-	// requirements.
+	// NodeName indicates in which node this pod is scheduled.
+	// If empty, this pod is a candidate for scheduling by the scheduler defined in schedulerName.
+	// If it is non-empty, the kubelet for this node becomes responsible for the lifecycle of this pod.
```
Comment (kannon92) [inline pkg/apis/core/types.go:3282]: Kubelet is always responsible for the lifecycle of the pod. If it is non empty, I thought it means that we skip the scheduling cycle for the pod.

Comment (alculquicondor) [reply pkg/apis/core/types.go:3282]: No to both.

The new comment is the accurate description.

Comment (liggitt) [reply pkg/apis/core/types.go:3282]: until nodeName is set, the kubelet isn't even aware of the pod's existence

once nodeName is set, kubelet takes over responsibility for running or rejecting the pod

Comment (alculquicondor) [reply pkg/apis/core/types.go:3282]: Let me give an example: it is possible to delete a Pod that is not scheduled, that is, a pod that has empty node name.

In that case, kubelet would have never looked at this Pod, even though it completed its lifecycle. In this case it's kcm which sets the Failed phase, if the Pod had a finalizer.

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/core/v1/types.go:3672]: A link out is fine, but *some* indication this is the wrong field to use if you want scheduling involved seems important

File: pkg/apis/core/types.go (hunk 2 above)
Comment (kannon92) [reply pkg/apis/core/types.go:3282]: Maybe I am taking "pod lifecycle" too literal. I was thinking https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ where this is primarly the responsibility of Kubelet. Kubelet always works with a set nodeName regardless of whether a user sets NodeName or not.

Comment (kannon92) [reply pkg/apis/core/types.go:3282]: I think I get what you are saying..

Comment (kannon92) [reply pkg/apis/core/types.go:3282]: SGTM.

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (alculquicondor) [reply staging/src/k8s.io/api/core/v1/types.go]: Added

**Pull request merged**